sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -311,6 +311,51 @@ namespace Core { namespace _internal {
      */
     ~PosixTimeConverter() = delete;
   };
+
+  /**
+   * @brief Provides convertion methods for Windows NT filetime to an #Azure::DateTime.
+   *
+   */
+  class NTFileTimeTimeConverter final {
+  public:
+    /**
+     * @brief Converts Windows NT filetime to an #Azure::DateTime.
+     *
+     * @param ntFiletime The number of 100-nanoseconds since 1601-01-01.
+     * @return Calculated #Azure::DateTime.
+     */
+    static DateTime NTFileTimeToDateTime(int64_t ntFiletime)
+    {
+      auto t = DateTime(1601) + Azure::_detail::Clock::duration(ntFiletime);
+      return DateTime(t);
+    }
+
+    /**
+     * @brief Converts a DateTime to Windows NT filetime.
+     *
+     * @param dateTime The `%DateTime` to convert.
+     * @return The number of 100-nanoseconds since 1601-01-01.
+     */
+    static int64_t DateTimeToNTFileTime(DateTime const& dateTime)
+    {
+      return std::chrono::duration_cast<Azure::_detail::Clock::duration>(dateTime - DateTime(1601))
+          .count();
```

Comment:
Will `count` always return 100-nanoseconds intervals? The doc doesn't seem to specify that:
https://cplusplus.com/reference/chrono/duration/count/

Reply:
Oh never mind, our `Clock::duration` is 100-nanoseconds (10^-7), which is what we are casting to.

A nit outside the scope of this PR, but a comment stating 10000000 is 100-ns would be useful to understand at a quick glance for readers, here:
https://github.com/Azure/azure-sdk-for-cpp/blob/9b29538f8a608d185186eff35a16ebfccad9672b/sdk/core/azure-core/inc/azure/core/datetime.hpp#L23-L24

Similar to:
https://github.com/Azure/azure-sdk-for-cpp/blob/9b29538f8a608d185186eff35a16ebfccad9672b/sdk/core/azure-core/inc/azure/core/datetime.hpp#L48-L49